sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRequest.java

```diff
@@ -14,9 +14,19 @@
 import java.net.URL;
 import java.nio.ByteBuffer;
 
+
 /**
- * The outgoing Http request. It provides ways to construct {@link HttpRequest} with {@link HttpMethod}, {@link URL},
- * {@link HttpHeader} and request body.
+ * Represents an outgoing HTTP request.
+ *
+ * <p>This class encapsulates an HTTP request, including the HTTP method, URL, headers, and body. It provides methods
+ * to set and get these properties.</p>
+ *
+ * <p>This class is useful when you want to create an HTTP request to send to a server. For example, you can use it to
+ * create a GET request to retrieve a resource, a POST request to create a resource, a PUT request to update a resource,
+ * or a DELETE request to delete a resource.</p>
```

Comment:
Instead of this, perhaps it is useful (as this is sort of one of the very core types of our Http stack) to explain how this is the abstraction for all requests through `azure.core.http` and related libraries?